src/writer/basic.rs

```diff
@@ -519,3 +534,48 @@ fn format_world<W: Debug>(world: Option<&W>, indent: usize) -> String {
         .then(|| format!("\n{}", world))
         .unwrap_or_default()
 }
+
+/// Formats [`gherkin::Table`], then adds `indent`s to each
+/// line to prettify the output.
+fn format_table(table: Option<&gherkin::Table>, indent: usize) -> String {
```

Comment:
It's better to accept just `&gherkin::Table` and instead
```rust
format_table(step.table.as_ref(), self.indent)
```
have
```rust
step.table
    .as_ref()
    .map(|t| format_table(t, self.indent))
    .unwrap_or_default()
```
So the function has a single responsibility.